lib/src/requests.dart

```diff
@@ -288,20 +288,20 @@ class Requests {
       json: json,
       body: body,
       queryParameters: queryParameters,
-      headers: headers,
+      headers: headers ?? const <String, String>{},
       timeoutSeconds: timeoutSeconds,
       persistCookies: persistCookies,
       verify: verify,
     );
   }
 
   static Future<Response> _httpRequest(HttpMethod method, String url,
-      {dynamic json,
-      dynamic body,
+      {dynamic? json,
+      dynamic? body,
       RequestBodyEncoding bodyEncoding = DEFAULT_BODY_ENCODING,
-      Map<String, dynamic> queryParameters,
-      int port,
-      Map<String, String> headers,
+      Map<String, dynamic>? queryParameters,
+      int? port,
+      required Map<String, String> headers,
```

Comment:
I suggest making headers nullable instead, since if you check under this function, it is reassigned by function _constructRequestHeaders, which also takes care of the case where headers is null